### ISSUE TITLE
Does not work with ng-submit in form fields

### ISSUE THREAD

**Issue by mooreds** [Siyfion/angular-typeahead#47]
If you use this in a form and use ng-submit to pull data from the ng-model object, the ng-model object is never set.  ng-click works fine.

Here's some code.  When using `ng-submit`, the `foo` model is never updated (so that `saveIt` can access it), whereas with `ng-click` it is.  However, with both the `{{}}` notation in the template works fine.

     // in controller
     $scope.saveIt = function() { 
        console.log($scope.foo)
     }

Below doesn't work:

     <form class="form-horizontal" role="form" ng-submit="saveIt()">
     // more ... form stuff
        <input id="hoods" type="text" class="sfTypeahead" options="exampleOptions" datasets="exampleData" ng-model="foo">
     <input type="submit" class="btn btn-primary" value="Save"/>

Below works:

    <form class="form-horizontal" role="form">
     // more ... form stuff
        <input id="hoods" type="text" class="sfTypeahead" options="exampleOptions" datasets="exampleData" ng-model="foo">
     <button ng-click="saveIt()"  class="btn btn-primary">Save</button>

**Comment by danmademe** [comment]
did this ever get solved?

**Comment by mooreds** [comment]
I don't believe so, but haven't checked it since the fall.

**Comment by devourment77** [comment]
I am running into the same issue.  My ng-model var just shows undefined in my form processing code when using ng-submit on the form tag.

**Comment by Siyfion** [comment]
I'd love to investigate this, but I just don't have any time free at the moment. Especially as I'm no longer even working with Angular, having switched to Meteor a while ago!

**Comment by RongBranovate** [comment]
this is still an issue, any idea how to get this resolved?

**Comment by RongBranovate** [comment]
This happens even if i send the `ng-model` value (which is set on the input) is being sent in the ng-submit function on the `form` element.
to show on the examples above:

```javascript
<form class="form-horizontal" role="form" ng-submit="saveIt(foo)">
 // more ... form stuff
    <input id="hoods" type="text" class="sfTypeahead" ng-model="foo" options="exampleOptions" datasets="exampleData">
 <input type="submit" class="btn btn-primary" value="Save"/>
```

I've solved this with jQuery at the moment but that's just because i already use it, if you don't then its not a good solution for you (you might want to use native JS to catch the element by id and call the submit function you want by yourself).

**Comment by hmil** [comment]
Seems fixed in v1.0.0. Please open a new issue if you still run into this bug.
